torch/nested/_internal/ops.py

```diff
@@ -21,11 +22,13 @@ def _outer_to_inner_dim(ndim, dim):
     return 0 if dim < 2 else dim - 1
 
 
-def _wrap_jagged_dim(ndim, dim, op_name, convert_to_inner_dim=True):
+def _wrap_jagged_dim(
+    ndim, dim, op_name, convert_to_inner_dim=True, allow_dim_zero=False
+):
     from torch._prims_common import canonicalize_dims
 
     wrapped = canonicalize_dims(ndim, dim)
-    if wrapped < 2:
+    if wrapped == 1 or (wrapped == 0 and not allow_dim_zero):
```

Comment:
nit: if we're going to do this, we should make sure the error message no longer mentions dim=0 since that's supported now